### PR TITLE
AppVeyor must be failed when pytest has any test failures.

### PR DESCRIPTION
**Pull request by shimizukawa**
### Feature or Bugfix
- Fix AppVeyor (Windows) CI status

### Purpose
- AppVeyor must be failed when pytest has any test failures.
- fix setup.py build_sphinx error on Windows.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ test_script:
       if (-not $tests) { $tests = '' }
       & "C:\Python$($env:PYTHON)\python.exe" -m pytest $test_ignore.Split(' ') --junitxml .junit.xml $tests.Split(' ')
       Pop-Location
-      if ($LastExitCode -eq 1) { Write-Host "Test Failures Occurred, leaving for test result parsing" }
```
Comment (shimizukawa) [inline .appveyor.yml:38]: AppVeyor must be failed when pytest has any test failures.

File: sphinx/io.py [hunk 2]
```diff
@@ -316,7 +316,8 @@ def read_doc(app, env, filename):
         pub = Publisher(reader=reader,
                         parser=parser,
                         writer=SphinxDummyWriter(),
-                        source_class=SphinxFileInput)
+                        source_class=SphinxFileInput,
+                        destination=NullOutput())
```
Comment (shimizukawa) [inline sphinx/io.py:320]: fix setup.py build_sphinx error on Windows.
https://ci.appveyor.com/project/sphinxdoc/sphinx/builds/21581633/job/lgs76s5ekk85ygmd#L2338

**Review by shimizukawa — COMMENTED** [review]
please review this.

File: sphinx/io.py (hunk 2 above)
Comment (tk0miya) [reply sphinx/io.py:320]: OMG... It's my wrong... I'm sorry and very thanks!

**Review by tk0miya — APPROVED** [review]

**Comment by shimizukawa** [comment]
Thanks for your reviewing!!!!

**Pull request merged**